### PR TITLE
Allow customization breadcrumb data of ActiveSupportLogger

### PR DESCRIPTION
**Pull request by igordepolli**
Ref: #2135 

## Description
Add new `config.rails.active_support_logger_subscription_items` to allow customization breadcrumb data of active support logger

```rb
config.rails.active_support_logger_subscription_items["sql.active_record"] << :type_casted_binds
config.rails.active_support_logger_subscription_items.delete("sql.active_record")
config.rails.active_support_logger_subscription_items["foo"] = :bar
```

### CODE REVIEW

File: sentry-rails/lib/sentry/rails/breadcrumb/active_support_logger.rb [hunk 1]
```diff
@@ -2,72 +2,12 @@ module Sentry
...
-          "analyze.active_storage" => %i[analyzer],
-        }.freeze
-
         class << self
           def add(name, started, _finished, _unique_id, data)
             # skip Rails' internal events
             return if name.start_with?("!")
 
-            allowed_keys = ALLOWED_LIST[name]
+            allowed_keys = Sentry.configuration.rails.active_support_logger_subscription_items[name]
```
Comment (sl0thentr0py) [inline sentry-rails/lib/sentry/rails/breadcrumb/active_support_logger.rb:10]: can you pass in the configuration list from `inject` like this? there are some multi-threaded edge cases with accessing `Sentry.configuration` that this avoids.

```patch
➜  sentry-rails git:(igordepolli/master) ✗ git --no-pager diff
diff --git a/sentry-rails/lib/sentry/rails/breadcrumb/active_support_logger.rb b/sentry-rails/lib/sentry/rails/breadcrumb/active_support_logger.rb
index 3af825d2..9dfdd1d2 100644
--- a/sentry-rails/lib/sentry/rails/breadcrumb/active_support_logger.rb
+++ b/sentry-rails/lib/sentry/rails/breadcrumb/active_support_logger.rb
@@ -7,10 +7,8 @@ module Sentry
             # skip Rails' internal events
             return if name.start_with?("!")

-            allowed_keys = Sentry.configuration.rails.active_support_logger_subscription_items[name]
-
             if data.is_a?(Hash)
-              data = data.slice(*allowed_keys)
+              data = data.slice(*@allowed_keys[name])
             end

             crumb = Sentry::Breadcrumb.new(
@@ -21,7 +19,9 @@ module Sentry
             Sentry.add_breadcrumb(crumb)
           end

-          def inject
+          def inject(allowed_keys)
+            @allowed_keys = allowed_keys
+
             @subscriber = ::ActiveSupport::Notifications.subscribe(/.*/) do |name, started, finished, unique_id, data|
               # we only record events that has a started timestamp
               if started.is_a?(Time)
diff --git a/sentry-rails/lib/sentry/rails/railtie.rb b/sentry-rails/lib/sentry/rails/railtie.rb
index 4e04d0e3..f77a2b7e 100644
--- a/sentry-rails/lib/sentry/rails/railtie.rb
+++ b/sentry-rails/lib/sentry/rails/railtie.rb
@@ -89,7 +89,7 @@ module Sentry
     def inject_breadcrumbs_logger
       if Sentry.configuration.breadcrumbs_logger.include?(:active_support_logger)
         require 'sentry/rails/breadcrumb/active_support_logger'
-        Sentry::Rails::Breadcrumb::ActiveSupportLogger.inject
+        Sentry::Rails::Breadcrumb::ActiveSupportLogger.inject(Sentry.configuration.rails.active_support_logger_subscription_items)
       end

       if Sentry.configuration.breadcrumbs_logger.include?(:monotonic_active_support_logger)
```

**Review by sl0thentr0py — CHANGES_REQUESTED** [review]
thx @igordepolli!
one small change then good to go

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,15 @@
 ## Unreleased
 
+### Features
+
+- Add new `config.rails.active_support_logger_subscription_items` to allow customization breadcrumb data of active support logger
```
Comment (sl0thentr0py) [inline CHANGELOG.md:5]:
```suggestion
- Add new `config.rails.active_support_logger_subscription_items` to allow customization breadcrumb data of active support logger [#2139](https://github.com/getsentry/sentry-ruby/pull/2139)
```

**Comment by igordepolli** [comment]
> thx @igordepolli! one small change then good to go

@sl0thentr0py Done. Ty for review :)

File: sentry-rails/lib/sentry/rails/configuration.rb [hunk 3]
```diff
@@ -80,6 +141,8 @@ class Configuration
       # ```
       attr_accessor :assets_regexp
 
+      attr_accessor :active_support_logger_subscription_items
```
Comment (st0012) [inline sentry-rails/lib/sentry/rails/configuration.rb:144]: Could you add some docs for it?

File: sentry-rails/spec/sentry/rails/breadcrumbs/active_support_logger_spec.rb [hunk 4]
```diff
@@ -65,6 +65,27 @@
...
+        {
+          "action" => "exception",
+          "params" => { "controller" => "hello", "action" => "exception" },
+          "format" => "html",
+          "method" => "GET", "path" => "/exception",
+        }
+      )
+      expect(breadcrumb["data"].keys).not_to include("controller")
+
+      Sentry.configuration.rails.active_support_logger_subscription_items["process_action.action_controller"] << :controller
```
Comment (st0012) [inline sentry-rails/spec/sentry/rails/breadcrumbs/active_support_logger_spec.rb:87]: You need to add this in a `ensure` block so it doesn't affect other tests if the test failed.

**Comment by igordepolli** [comment]
@st0012 Done. Good catch! Ty! :)

**Review by sl0thentr0py — APPROVED** [review]

**Comment by sl0thentr0py** [comment]
@igordepolli the `ensure` fails on ruby 2.4 because this feature was added in 2.5, you can either add a `begin/end` or just wrap the spec in a `context` with a `before/after` which is probably cleaner.

https://github.com/getsentry/sentry-ruby/actions/runs/6620176438/job/17999733560?pr=2139#step:5:156

You can also just give me [push rights to the PR](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) and I'll take care of the rest and merge this in.

**Comment by igordepolli** [comment]
> @igordepolli the `ensure` fails on ruby 2.4 because this feature was added in 2.5, you can either add a `begin/end` or just wrap the spec in a `context` with a `before/after` which is probably cleaner.
> 
> https://github.com/getsentry/sentry-ruby/actions/runs/6620176438/job/17999733560?pr=2139#step:5:156
> 
> You can also just give me [push rights to the PR](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) and I'll take care of the rest and merge this in.

@sl0thentr0py Oh, sorry for the inconvenience. You already have the right to manipulate PR however you want, feel free.

**Pull request merged**